src/main/scala/com/ubirch/controllers/AcctEventsController.scala

```diff
@@ -63,39 +64,86 @@ class AcctEventsController @Inject() (
+          ownerId <- Task(rawOwnerId)
             .map(_.map(UUID.fromString).get) // We want to know if failed or not as soon as possible
-            .onErrorHandle(_ => throw InvalidParamException("Invalid OwnerId", "Wrong owner param"))
+            .onErrorHandle(_ => throw new IllegalArgumentException("Invalid OwnerId: wrong owner param: " + rawOwnerId.getOrElse("")))
 
-          ownerCheck = token.ownerIdAsUUID.map(_ == ownerId).isSuccess || (token.ownerIdAsUUID.map(_ != ownerId).isSuccess && token.isAdmin)
-          _ = earlyResponseIf(ownerCheck)(InvalidSecurityCheck("Invalid Owner Relation", "You can't access somebody else's data"))
+          ownerIdFromToken <- Task.delay(token.ownerIdAsUUID.getOrElse(throw new IllegalArgumentException("Invalid Token OwnerI: Wrong token owner")))
+
+          ownerCheck = (ownerIdFromToken == ownerId) || token.isAdmin
+          _ <- earlyResponseIf(!ownerCheck)(InvalidSecurityCheck("Invalid Owner Relation", "You can't access somebody else's data"))
+
+          cat <- Task(params.get("cat"))
+            .map(_.filter(_.nonEmpty))
+            .onErrorHandle(_ => throw new IllegalArgumentException("Invalid cat: wrong cat param"))
 
           identityId <- Task(params.get("identity_id"))
             .map(_.map(UUID.fromString))
-            .onErrorHandle(_ => throw InvalidParamException("Invalid identity_id", "Wrong identity_id param"))
-
-          evs <- acctEvents.byOwnerIdAndIdentityId(ownerId, identityId)
-            .toListL
+            .onErrorHandle(_ => throw new IllegalArgumentException("Invalid identity_id: wrong identity_id param"))
+
+          start <- Task(params.get("start"))
+            .map(_.map(sdf.parse))
+            .map(_.map(x => DateUtil.dateToLocalTime(x)))
+            .onErrorHandle(_ => throw new IllegalArgumentException("Invalid Start: Use yyyy-M-dd this format"))
+
+          end <- Task(params.get("end"))
+            .map(_.map(sdf.parse))
+            .map(_.map(x => DateUtil.dateToLocalTime(x)))
+            .onErrorHandle(_ => throw new IllegalArgumentException("Invalid End: Use yyyy-M-dd this format"))
+
+          _ <- earlyResponseIf(start.isDefined && end.isEmpty)(new IllegalArgumentException("Invalid Range Definition: Start requires End"))
+          _ <- earlyResponseIf(start.isEmpty && end.isDefined)(new IllegalArgumentException("Invalid Range Definition: End requires Start"))
+          _ <- earlyResponseIf({
+            for {
+              s <- start
+              e <- end
+            } yield s.isAfter(e)
+          }.getOrElse(false))(new IllegalArgumentException("Invalid Range Definition: Start must be before End"))
+
+          paramsString = s"only_count->$onlyCount, bucketed->$bucketed, owner_id->$ownerId, cat=${cat.getOrElse("")}, identity_id->${identityId.getOrElse("")}, start=${start.getOrElse("")}, end=${end.getOrElse("")}"
+          _ = logger.info(s"query = $paramsString")
+
+          evs <- if (onlyCount) {
```

Comment:
Instead of taking these parameters(only_count, bucketed) separately, a single parameter that specifies the query mode makes more sense to me.

Reply:
Ok, sounds like a good idea. Added